### DISCUSSION TITLE
"Pause and Discuss" proposed changes

### DISCUSSION THREAD

**Discussion by arimxyer** [cline/cline#724] [Feature Requests]
I'm not sure about everyone else, but I'm constantly asking cline questions on code that it's made changes to - like whether or not that change fits the apps current designed framework and flow?, or what the logic behind making the certain change is?, etc., and whenever I do ask the proposed file changes go away which I understand is normal behavior in this instance(which sometimes I don't want them too, mainly in fear of it re-attempting to make the file changes and not making the same adjustments). 

Would be great if there was an additional button next to the "Accept" & "Reject" buttons to "Pause and Discuss", so we could ask questions about the proposed edits it's attempting to make, and potentially understand the LLM's line of reasoning or thinking, or to adjust the direction and focus of its suggested adjustment.

**Comment by dancingmadkefka** [comment]
Was just writing this exact same feature request and then did a differently-worded search once more before submitting and found this :)

It's also a matter of clogging up the context window with file-reads each time it has to throw away its proposed changes and then read the file again and propose more. Would indeed be great to be able to just discuss them instead.

Also would be great to be able to add comments WHILE saving. Typing a message and hitting send is classed as a rejection, but typing and hitting save is the same as just hitting save. This is an easy one to implement too I think.

**Reply by Jdo300** [reply]
YES. Please Add!

**Comment by chrisparkernz** [comment]
Something like this would be incredibly useful.
